libs/SmartSync/SmartSyncTests/SyncManagerTests.m

```diff
@@ -269,37 +268,43 @@ - (void)testCleanResyncGhostsForMRUTarget
 - (void)testCleanResyncGhostsForSOSLTarget
 {
 
+    [self createAccountsSoup:ACCOUNTS_SOUP];
+
     // Creates 1 account on the server.
-    NSMutableDictionary* accountIdToFields = [[NSMutableDictionary alloc] initWithDictionary:[self createAccountsOnServer:1]];
-    XCTAssertEqual(accountIdToFields.count, 1, @"1 account should have been created");
+    NSDictionary* accountIdToFields = [self createAccountsOnServer:1];
+    [NSThread sleepForTimeInterval:1]; //give server a second to settle to reflect in API
+
     NSArray* accountIds = [accountIdToFields allKeys];
-    NSString* soupName = ACCOUNTS_SOUP;
-    [self createAccountsSoup:soupName];
+    
+    NSMutableArray* accountNames = [NSMutableArray new];
+    for (NSDictionary* fields in [accountIdToFields allValues]) {
+        [accountNames addObject:fields[NAME]];
+    }
 
     // Builds SOSL sync down target and performs initial sync.
-    SFSDKSoslBuilder* soslBuilder = [SFSDKSoslBuilder withSearchTerm:accountIdToFields[accountIds[0]][NAME]];
+    NSString* searchQuery = [accountNames componentsJoinedByString:@" OR "];
+    SFSDKSoslBuilder* soslBuilder = [SFSDKSoslBuilder withSearchTerm:searchQuery];
     SFSDKSoslReturningBuilder* returningBuilder = [SFSDKSoslReturningBuilder withObjectName:ACCOUNT_TYPE];
     [returningBuilder fields:@"Id, Name, Description"];
     NSString* sosl = [[[soslBuilder returning:returningBuilder] searchGroup:@"NAME FIELDS"] build];
-    NSNumber* syncId = [NSNumber numberWithInteger:[self trySyncDown:SFSyncStateMergeModeLeaveIfChanged target:[SFSoslSyncDownTarget newSyncTarget:sosl] soupName:soupName totalSize:accountIdToFields.count numberFetches:1]];
-    SFQuerySpec *querySpec = [SFQuerySpec newAllQuerySpec:soupName withOrderPath:ID withOrder:kSFSoupQuerySortOrderAscending withPageSize:10];
-    NSUInteger numRecords = [store countWithQuerySpec:querySpec error:nil];
-    XCTAssertEqual(numRecords, 1, @"1 account should be stored in the soup");
+    NSNumber* syncId = [NSNumber numberWithInteger:[self trySyncDown:SFSyncStateMergeModeLeaveIfChanged target:[SFSoslSyncDownTarget newSyncTarget:sosl] soupName:ACCOUNTS_SOUP totalSize:accountIds.count numberFetches:1]];
+    [self checkDbExists:ACCOUNTS_SOUP ids:accountIds idField:@"Id"];
```

Comment:
Instead of counting before and after, we check that the record is actually gone. (The same change was done for Android).